services/common/src/components/comments/CommentEditor.tsx

```diff
@@ -0,0 +1,72 @@
+import React, { FC } from "react";
+import { useSelector } from "react-redux";
+import { Button, Form, Input } from "antd";
+import { userHasRole } from "@mds/common/redux/reducers/authenticationReducer";
+
+interface CommentEditorProps {
+  onSubmit: (data: any) => void;
+  addCommentPermission: string;
+  maxLength?: number;
+}
+
+export const CommentEditor: FC<CommentEditorProps> = ({
+  onSubmit,
+  addCommentPermission,
+  maxLength = 300,
+}) => {
+  const [submitting, setSubmitting] = React.useState(false);
+  const [comment, setComment] = React.useState("");
+
+  const handleReset = () => {
+    setComment("");
+    setSubmitting(false);
+  };
+
+  const handleSubmit = async (event) => {
+    event.preventDefault();
+    setSubmitting(true);
+
+    await onSubmit({ comment });
+    handleReset();
+  };
+
+  const handleChange = (e) => {
+    setComment(e.target.value);
+  };
+
+  const canAddComment = addCommentPermission
+    ? useSelector((state) => userHasRole(state, addCommentPermission))
```

Comment:
I do believe that putting a hook in a conditional statement is a hooks rules violation (gotta separate it out into its own statement)